aicsimageio/readers/reader.py

```diff
@@ -121,45 +144,74 @@ def set_scene(self, id: int):
         ----------
         id: int
             The scene id to set as the operating scene.
-        """
-        pass
 
-    @staticmethod
-    @abstractmethod
-    def _assert_reader_supports_image(image: types.ImageLike) -> bool:
-        """
-        The per-Reader implementation of validating that an image is supported or not by
-        the Reader itself.
+        Raises
+        ------
+        IndexError: the provided scene id is not found in the available scene id list
         """
         pass
 
     @abstractmethod
-    def _read_delayed(self) -> da.Array:
+    def _read_delayed(self) -> xr.DataArray:
         """
-        The delayed dask array constructor for the image.
+        The delayed data array constructor for the image.
 
         Returns
         -------
-        dask_data: da.Array
-            The fully constructed delayed dask array.
+        data: xr.DataArray
+            The fully constructed delayed DataArray.
 
             It is additionally recommended to closely monitor how dask array chunks are
             managed.
         """
         pass
 
     @abstractmethod
-    def _read_immediate(self) -> np.ndarray:
+    def _read_immediate(self) -> xr.DataArray:
```

Comment:
I'd feel better about this if I could see an implemented _read_immediate for xr_DataArray